library/src/main/java/org/apache/apex/malhar/lib/wal/FileSystemWAL.java

```diff
@@ -680,9 +681,7 @@ protected void flush() throws IOException
           //until the stream is closed on the local FS, readers don't see any data.
           close();
         } else {
-          Syncable syncableOutputStream = (Syncable)outputStream;
-          syncableOutputStream.hflush();
-          syncableOutputStream.hsync();
+          ((HdfsDataOutputStream)outputStream).hsync(EnumSet.of(SyncFlag.UPDATE_LENGTH));
```

Comment:
I have updated the HDFS library version.

Please refer to following link
http://stackoverflow.com/questions/32231105/why-is-hsync-not-flushing-my-hdfs-file

Just doing hsync won't update the file length with NameNode. We have different options of implementing the solution,

1. Close & re-open the wal file, this will update the length with NameNode and no other changes are required.

2. Don't depend on the length provided by the fileContext, as mentioned in the SO link

3. Pass the flag while calling HSync ( no other changes ). 

Earlier versions of the HDFS WAL could have used one of the first 2 options.